ROADMAP.md

```diff
@@ -0,0 +1,137 @@
+
+----------------------------------------------------------------
+
+# Milestone 2 - Improve Connectivity of go-libp2p/go-ipfs
+
+> Summary: Improve the connectivity of go-libp2p. Have the technical discussions that will enable us to follow a plan to achieve a better connected graph.
+
+### Tasks:
+
+- [ ] Technical Discussion: Define a strategy for connection closing
+- [ ] Implement the strategy for connection closing in go-libp2p
+- [ ] Technical Discussion: Definte a strategy for Relay (or multiple Relay)
+- [ ] Implement Relay in go-libp2p
+- [ ] Technical Discussion: Improve NAT Traversal
+- [ ] Improve NAT traversal in go-libp2p
+
+### Dependencies:
+
+### Requirements by other projects:
+
+### Notes:
+
+##### Expected date of completion: `Q3`
+
+----------------------------------------------------------------
+
+# Milestone 3 - Developer Experience
+
+> Summary: The main goal of this milestone is to get more developers excited about libp2p and learn how to use it.
+
+### Tasks:
+
+- [ ] Replicate IPFS community guidelines for the libp2p project
+- [ ] libp2p.io
+- [ ] Examples
+  - [ ] How to use js-libp2p (dial, multi transport, stream muxing, identify, protocol multiplexing)
+  - [ ] How to use go-libp2p (everything js-libp2p + DHT)
+- [ ] Spec
+  - [ ] Move spec to the libp2p org
+  - [ ] Revisit and improve
+- [ ] interface-libp2p
+  - [ ] Make js-libp2p-ipfs and js-libp2p-ipfs-browser use it
+  - [ ] Create js-libp2p, a base class that other libp2p builds can reuse
+
+### Dependencies:
+
+### Requirements by other projects:
+
+### Notes:
+
+##### Expected date of completion: `Q3 + Q4`
+
+----------------------------------------------------------------
+
+# Milestone 4 - Move DHT to use IPRS
+
+> Summary: Currently our records are simple protobufs that get stored in the datastore. With this milestone, any record (provider or IPNS) will be a full IPLD object that follows the IPRS spec.
+
+### Tasks:
+
+- [ ] Technical Discussion: Decide if whether we should support two types of records or simply create another universe of the DHT
+- [ ] Migrate go-ipfs to use IPRS
+- [ ] js-libp2p-dht (also known as dht legacy)
+
+### Dependencies:
+
+### Requirements by other projects:
+
+### Notes:
+
+##### Expected date of completion: `Q4`
+
+----------------------------------------------------------------
+
+# Milestone 5 - Break DHT into Peer Routing and Content Routing
+
+> Summary: 
+
+### Tasks:
+
+- [ ] Technical Discussion: Revisit proposed spec
+
+### Dependencies:
+
+### Requirements by other projects:
+
+### Notes:
+
+This milestone is still very green, however something we know for sure we want to do.
+
+
+##### Expected date of completion: `TBA`
+
+----------------------------------------------------------------
+
+# Milestone 6 - Unreliable Overlay Network Routing
+
+> Summary: The packet switched network
+
+### Tasks:
+
+### Dependencies:
+
+### Requirements by other projects:
+
+### Notes:
+
+##### Expected date of completion: `TBA`
```

Comment:
@lgierth could you fill in this milestone with tasks, tracking issues and notes you see for multigram, and overall packeted switched network? Thank you